modules/extensions/src/data-filter/shader-module.ts

```diff
@@ -6,33 +6,43 @@ import {glsl} from '../utils/syntax-tags';
  * data filter shader module
  */
 const vs = glsl`
-uniform DATAFILTER_TYPE filter_min;
-uniform DATAFILTER_TYPE filter_softMin;
-uniform DATAFILTER_TYPE filter_softMax;
-uniform DATAFILTER_TYPE filter_max;
 uniform bool filter_useSoftMargin;
 uniform bool filter_enabled;
 uniform bool filter_transformSize;
 uniform ivec4 filter_categoryBitMask;
 
-#ifdef NON_INSTANCED_MODEL
-  #define DATAFILTER_ATTRIB filterValues
-  #define DATAFILTER_ATTRIB_64LOW filterValues64Low
-  #define DATACATEGORY_ATTRIB filterCategoryValues
-#else
-  #define DATAFILTER_ATTRIB instanceFilterValues
-  #define DATAFILTER_ATTRIB_64LOW instanceFilterValues64Low
-  #define DATACATEGORY_ATTRIB instanceFilterCategoryValues
+#ifdef DATAFILTER_TYPE
+  uniform DATAFILTER_TYPE filter_min;
+  uniform DATAFILTER_TYPE filter_softMin;
+  uniform DATAFILTER_TYPE filter_softMax;
+  uniform DATAFILTER_TYPE filter_max;
+
+  #ifdef NON_INSTANCED_MODEL
+    #define DATAFILTER_ATTRIB filterValues
+    #define DATAFILTER_ATTRIB_64LOW filterValues64Low
+  #else
+    #define DATAFILTER_ATTRIB instanceFilterValues
+    #define DATAFILTER_ATTRIB_64LOW instanceFilterValues64Low
+  #endif
+
+  in DATAFILTER_TYPE DATAFILTER_ATTRIB;
+  #ifdef DATAFILTER_DOUBLE
+    in DATAFILTER_TYPE DATAFILTER_ATTRIB_64LOW;
+
+    uniform DATAFILTER_TYPE filter_min64High;
```

Comment:
Thought: I would have focused on just `#ifdef`-ing out the attributes, and not have worried about the uniforms. Based on the principle that less conditional code is better, so we should use it sparingly.

Also we should be moving towards uniform buffers and there is not much penalty for having unused fields in a uniform buffer (or even "global" uniforms).

Reply:
These need to be inside the `#ifdef` as their type is `DATAFILTER_TYPE`